flashlight/fl/examples/RnnClassification.cpp

```diff
@@ -0,0 +1,325 @@
+        linear_(Linear(hiddenSize, numClasses)),
+        logsoftmax_(0) {
+    std::cout << "Creating a RNN Classifier with vocab size: " << vocabSize
+              << " and num classes: " << numClasses << std::endl;
+    add(embed_);
+    add(rnn_);
+    add(linear_);
+    add(logsoftmax_);
+  }
+
+  std::vector<Variable> forward(const std::vector<Variable>& inputs) override {
+    throw std::runtime_error("Not implemented");
+  }
+
+  std::tuple<Variable, Variable, Variable>
+  forward(const Variable& input, const Variable& h, const Variable& c) {
+    const unsigned numChars = input.dims(1);
+    Variable ho, co; // hidden and carry output
+    // output should be hs x bs x ts : [ 𝑋𝑖𝑛, 𝑁, 𝑇 ]
+    Variable output = embed_(input);
+    // The input to the RNN is expected to be of shape [ 𝑋𝑖𝑛, 𝑁, 𝑇 ] where 𝑋𝑖𝑛
+    // is the hidden size, 𝑁 is the batch size and 𝑇 is the sequence length.
+    std::tie(output, ho, co) = rnn_(output, h, c);
+    // The output of the RNN should be of shape [ 𝑋𝑜𝑢𝑡, 𝑁, 𝑇], with 𝑋𝑜𝑢𝑡 to be
+    // the hidden_size (unidirectional RNN)
+    // Truncate BPTT
+    ho.setCalcGrad(false);
+    co.setCalcGrad(false);
+    output = linear_(output);
+    output = logsoftmax_(output);
+    output = output(af::span, af::span, input.array().dims(1) - 1);
+    return std::make_tuple(output, ho, co);
+  }
+
+  std::string prettyString() const override {
+    return "RnnClassifer";
+  }
+
+  // Inference on the given input: returns the category index
+  unsigned
+  infer(const std::string& inputString, const Variable& h, const Variable& c) {
+    af::array ia = ClassificationDataset::lineToTensor(inputString);
+    Variable output, ho, co;
+    std::tie(output, ho, co) = forward(noGrad(ia), h, c);
+    af::array maxValue, prediction;
+    af::max(maxValue, prediction, output.array(), 0);
+    unsigned classId = prediction.scalar<unsigned>();
+    return classId;
+  }
+
+  // Predict the category of the given input, compared with the expected label
+  // and print the result
+  bool unittest(const std::string& input, const std::string& expectedLabel) {
+    Variable output, h, c;
+    auto p = ClassificationDataset::Id2Label[infer(input, h, c)];
+    const std::string s = (p == expectedLabel ? "✓ " : "✗ ");
+    std::cout << "input: " << std::setw(20) << input
+              << "\t expected: " << expectedLabel << "\t prediction: " << p
+              << "\t" << s << std::endl;
+  }
+
+ private:
+  Embedding embed_;
+  RNN rnn_;
+  Linear linear_;
+  LogSoftmax logsoftmax_;
+};
+
+int main(int argc, char** argv) {
+  std::cout << "RnnClassification (path to the data dir) (learning rate) (num "
+               "epochs) (hiddensize)"
+            << std::endl;
+  std::cout << "Dataset : https://download.pytorch.org/tutorial/data.zip"
+            << std::endl;
+  if (argc < 2) {
+    std::cout << "To setup the dataset: " << std::endl;
+    std::cout << "wget https://download.pytorch.org/tutorial/data.zip") << std::endl;
```

Comment:
We are ready to land, but CI crashing because of ")". Could you fix this by removing ")" in the 226, 227 and 228 line?

Reply:
@WilliamTambellini

Reply:
done